runtime/src/stakes.rs

```diff
@@ -182,12 +191,19 @@ impl Stakes {
             }
 
             if account.lamports == 0 {
+                // when account is removed (lamports == 0), remove it from Stakes as well
+                // so that given `pubkey` can be used for any owner in the future, while not
+                // affecting Stakes.
                 self.stake_delegations.remove(pubkey);
             } else if let Some(delegation) = delegation {
                 self.stake_delegations.insert(*pubkey, delegation);
             }
             None
         } else {
+            // there is no need to remove possibly existing Stakes cache entries with given
```

Comment:
@CriesofCarrots `Account::default()` was hitting here in #14062.

Reply:
💯